core/capabilities/launcher.go

```diff
@@ -118,6 +118,98 @@ func NewLauncher(
 	}
 }
 
+// Maintain only necessary Don2Don connections:
+//   - Workflow DONs connect only to other DONs that have at least one remote capability
```

Comment:
@bolekk to support local DON capabilities, workflows DONs would need to interconnect, given this maybe this change be put on hold for now?

Reply:
I don't follow. Local capabilities are local by definition so we don't need don2don.

Reply:
Do you mean that nodes from a given workflow DON still need to be connected via don2don to other nodes from the same DON? I'm not convinced that it will be necessary. Let's ignore it for now and we can re-add later when (if?) the remote shims are modified to support it.